extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -232,7 +228,6 @@ CamelMainBuildItem main(
                 CamelContextBuildItem context,
                 CamelRoutesCollectorBuildItem routesCollector,
                 List<CamelMainListenerBuildItem> listeners,
-                List<CamelRoutesBuilderBuildItem> routesBuilders,
```

Comment:
I should be able to add routes that are instantiated at build time, i.e. if I use camel annotation for DI I do not need ArC at all thus I can optimize a bit my application.

Reply:
That would re-introduce the double initialization of the routes.

I was thinking of filtering by `@ApplicationScoped` that can be done well using Jandex: Builders having `@ApplicationScoped` would behave like in the current proposal and all others could go via `CamelRoutesBuilderBuildItem`. This would be more complex and thus potentially more buggy and harder to maintain. Maybe you have a better idea?

Reply:
Camel has the ability to do class-path scanning to discover routes which can be activated by configuring which packages camel has to scan for `RoutesBuilder` classes (already supported by other platform such as Spring, CDI, Main) so we can mimic that behaviour at build time with a `quarkus.camel.` option (note that this can activated also right now through camel main properties but class path scanning would happen at runtime).

By doing that, routes builders can be added to the context by:
- producing a `CamelBeanBuildItem`
- using a `MainListener`
- using a CDI annotation such as `@ApplicationScoped`
- listing the packages to scan for auto discovery (as far as I remember discovered routes have lower precedence than those available from the container thus discarded if the same route class is also provided by the container)

An additional option that has been discussed would be to leverage camel's annotation `@BindToRegistry` which would act like a cross platform lite DI.

Reply:
Yes, I agree. You perhaps meant CamelRoutesBuilderBuildItem rather than CamelBeanBuildItem?

The ultimate aim of this PR is to make the `@Inject`, `@ConfigProperty`, etc. work in RoutesBuilders (see the tests in this PR that currently fail). I agree that the problem of double Route initialization is actually the root cause. 

My previous proposal to solve both problems by delegating the creation of all RouteBuilders to Arc was rejected for boot performance reasons which I found quite legitimate. 

I proposed an alternative approach where Arc beans are thrown away from the list provided via CamelRoutesBuilderBuildItem, but I see no feedback for that. 

In between, I looked how to implement that reliably, and found that the list of Arc beans could perhaps be pulled from Arc itself like this: https://github.com/ppalaga/camel-quarkus/commit/cdbfd1999b35e103944d41e570d266d73fd20995#diff-ec6205752018e84a80fcd9eaa92af189R203-R209 (once they change the visibility of some methods, I asked the Arc guys about that). What do you think about the linked solution?

Reply:
No CamelBeanBuildItem is - IMHO - the right bean to use as every route bound to the registry is then automatically discovered so CamelRoutesBuilderBuildItem is superfluous.

The problem about using ArC in STATIC_INIT is that it may not be aware of the "runtime beans" at that point in time, at least in my experience.